cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/kubernetes/AbstractNamespacedResourceOperatorTest.java

```diff
@@ -117,17 +123,40 @@ protected T modifiedResource()  {
     /** Create the subclass of ResourceOperation to be tested */
     protected abstract AbstractNamespacedResourceOperator<C, T, L, R> createResourceOperations(Vertx vertx, C mockClient);
 
-    /** Create the subclass of ResourceOperation to be tested with mocked readiness checks*/
-    protected AbstractNamespacedResourceOperator<C, T, L, R> createResourceOperationsWithMockedReadiness(Vertx vertx, C mockClient)    {
+    /**
+     *  Create the subclass of ResourceOperation to be tested.
+     *  The boolean parameter useServerSideApply is not used by default.
+     *  In case that resource supports SSA, the method should be overridden in resource's test class
+     */
+    protected AbstractNamespacedResourceOperator<C, T, L, R> createResourceOperations(Vertx vertx, C mockClient, boolean useServerSideApply) {
         return createResourceOperations(vertx, mockClient);
     }
 
-    @Test
-    public void testCreateWhenExistsWithChangeIsAPatch(VertxTestContext context) {
-        testCreateWhenExistsWithChangeIsAPatch(context, true);
+    /** Create the subclass of ResourceOperation to be tested with mocked readiness checks*/
+    protected AbstractNamespacedResourceOperator<C, T, L, R> createResourceOperationsWithMockedReadiness(Vertx vertx, C mockClient, boolean useServerSideApply)    {
+        return createResourceOperations(vertx, mockClient, useServerSideApply);
     }
 
-    public void testCreateWhenExistsWithChangeIsAPatch(VertxTestContext context, boolean cascade) {
+    /**
+     * Returns combinations for parameterized test cases for Server Side Apply.
+     * The default is only combination - with ServerSideApply disabled.
+     * Resources that support both options has to override the methods with combination of true and false.
+     *
+     * @return  combinations for parameterized test cases for Server Side Apply.
+     */
+    protected Stream<Arguments> useServerSideApplyCombinations() {
```

Comment:
This should probably have more prominent position in the order of the methods? E.g. right after or before the afterall / beforeall methods etc.? It should not be just lost like this somewhere in the middle.

Reply:
I can move it, the only two reasons I left it before the first test are that when you see the method source, it's right above the first parametrized test and the second one is that it's like that in other test classes as well. That's why I thought it will make sense to have it like this.

Anyway, I will move it a bit up after the before/after methods.